### PR TITLE
Replace knox with aws-sdk

### PR DESCRIPTION
**Pull request by nimishparmar**
@ali ptal

### CODE REVIEW

File: uploader-test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+This is not a proper test. Only used as proof of concept.
```
Comment (nimishparmar) [inline uploader-test.js]: Please DO NOT merge this test. I left it here as a reference. Once everything look OK on index.js, I'll remove this file before the final merge :)

File: index.js [hunk 2]
```diff
@@ -106,43 +113,53 @@ var S3Sync = klass(function (config, options) {
...
-          )
-          .on('error', function (e) {
-            console.error('error in putting new file', e)
+          var md5body =  fs.createReadStream(file)
+          var md5headers = _.extend({}, this.getSettings(), this._mergeHeaders(md5File))
+          this.client.upload({
+            Body: md5body,
+            Bucket: this.bucket,
+            Key: md5File,
+            md5headers
```
Comment (alif) [inline index.js]: this will result in an object:
```

Comment (nimishparmar) [inline index.js]: Yes. These are the headers for the md5file. Sample value of the `md5headers` variable is `{ 'x-amz-acl': 'public-read', 'Content-Type': 'image/png' }`

**Comment by nimishparmar** [comment]
Removed the temp test file. I linked my local version of `s3-asset-uploader` module to `tasks/upload-assets.js` and called `make assets` target with `nimish-s3-asset-uploader` bucket. Everything worked and I saw all the assets being uploaded to that bucket. After uploading all the assets, the digest json file was also created. Once everything was finished, the lock was released cleanly and the process exited with code 0.

File: index.js (hunk 2 above)
Comment (alif) [inline index.js]: I mean this will result in an object with an `md5headers` attribute:

Comment (alif) [inline index.js]: Also, the expected attributes for `acl` & `content-type` are `ACL` and `ContentType`. I'll make these changes.

Comment (alif) [inline index.js]: http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property

File: index.js [hunk 3]
```diff
@@ -66,7 +68,7 @@ var S3Sync = klass(function (config, options) {
         this.writeDigestFile(function (err, resp) {
           debug('wrote digest file')
           this.options.complete && this.options.complete()
-        })
+        }.bind(this))
```
Comment (alif) [inline index.js]: I bet this is the bug that broke things for 0.6-0.8

File: index.js (hunk 2 above)
Comment (nimishparmar) [inline index.js]: Hm, yes md5headers would be stored as an object. crap. I keep forgetting about lodash. Really need to keep that in mind whenever dealing with objects. _.merge is so handy.

File: index.js (hunk 3 above)
Comment (nimishparmar) [inline index.js]: Without this bind, the `complete` property wasn't getting called and the process hung around. Also the lock wasnt released. Was this the brokenness in those versions?

Comment (alif) [inline index.js]: Actually - nevermind L88 would have made sure the callback was bound to `this`

Comment (nimishparmar) [inline index.js]: Line 98 you mean? I thought so too, but without the bind on Line 71 it wasn't working.

**Pull request merged**